CRATE.md

```diff
@@ -0,0 +1,148 @@
+<p align="center">
+  <img src="https://raw.githubusercontent.com/nats-io/nats.rs/master/logo/NATS-Rust.png">
+</p>
+
+<p align="center">
+    A <a href="https://www.rust-lang.org/">Rust</a> client for the <a href="https://nats.io">NATS messaging system</a>.
+</p>
+
+## Status
+[![License Apache 2](https://img.shields.io/badge/License-Apache2-blue.svg)](https://www.apache.org/licenses/LICENSE-2.0)
+[![Crates.io](https://img.shields.io/crates/v/nats.svg)](https://crates.io/crates/nats)
+[![Documentation](https://docs.rs/nats/badge.svg)](https://docs.rs/nats/)
+[![Build Status](https://travis-ci.org/nats-io/nats.rs.svg?branch=master)](http://travis-ci.org/nats-io/nats.rs)
+
+Warning: as of version 0.4.0, this crate has been rewritten, and several public
+APIs have changed. Please see the [docs](https://docs.rs/nats), the [new
+repo](https://github.com/nats-io/nats.rs), and [the
+changelog](https://github.com/nats-io/nats.rs/blob/master/CHANGELOG.md) for
+details.
+
+## Motivation
+Rust may be the most interesting new language the NATS ecosystem has seen. We
+believe this client will have a large impact on NATS, distributed systems, and
+embedded and IoT environments. With Rust we wanted to be as idiomatic as we
+could be and embrace the language. We moved many things that would have been
+runtime checks and errors to the compiler, most notably options on connections,
+and having subscriptions generate multiple styles of iterators, since iterators
+are a first class citizen in Rust. We also wanted to be aligned with the NATS
+philosophy of simple, secure, and fast! (The security is coming soon!)
+
+## Feedback
+
+This is a new client, and I and the team are new to Rust, so all feedback welcome!
+We encourage all folks in the NATS and Rust ecosystems to help us improve this library. Please open issues, submit PRs, etc.
+
+## Example Usage
```

Comment:
Any new ones we want to add?

Reply:
added auth and TLS constructor examples, fixed some compiler errors with some of the expamples

Reply:
NGS/Creds example?

Reply:
Also the "I am new to Rust" probably should be removed since you and Stjepan are more representative of the project now.